sys/shell/commands/sc_dfplayer.c

```diff
@@ -0,0 +1,617 @@
+/*
+ * Copyright 2019 Marian Buschsieweke
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     sys_shell_commands
+ * @{
+ *
+ * @file
+ * @brief       Provides a shell command to control a DFPlayer Mini
+ *
+ * @author      Marian Buschsieweke <marian.buschsieweke@ovgu.de>
+ *
+ * @}
+ */
+
+#include <errno.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "dfplayer.h"
+#include "dfplayer_internal.h"
+#include "fmt.h"
+
+static const char *_equalizers[] = {
+    [DFPLAYER_EQ_NORMAL]    = "normal",
+    [DFPLAYER_EQ_POP]       = "pop",
+    [DFPLAYER_EQ_ROCK]      = "rock",
+    [DFPLAYER_EQ_JAZZ]      = "jazz",
+    [DFPLAYER_EQ_CLASSIC]   = "classic",
+    [DFPLAYER_EQ_BASE]      = "base",
+    [DFPLAYER_EQ_NUMOF]     = NULL
+};
+
+static const char *_modes[] = {
+    [DFPLAYER_MODE_UNKOWN]      = "unknown",
+    [DFPLAYER_MODE_REPEAT_DIR]  = "repeat folder",
+    [DFPLAYER_MODE_REPEAT]      = "repeat",
+    [DFPLAYER_MODE_RANDOM]      = "random",
+    [DFPLAYER_MODE_NORMAL]      = "normal",
+    [DFPLAYER_MODE_NUMOF]       = NULL
+};
+
+static const char *_states[] = {
+    [DFPLAYER_STATE_PLAYING]    = "playing",
+    [DFPLAYER_STATE_PAUSED]     = "paused",
+    [DFPLAYER_STATE_STOPPED]    = "stopped",
+    [DFPLAYER_STATE_NUMOF]      = NULL
+};
+
+static void _print_error(int retval)
+{
+    print_str("Error: ");
+    print_str(strerror(-retval));
```

Comment:
looks like this is not in the AVR libc.

Reply:
That makes sense. On a Harvard arch, this would result in a lot of error messages being allocated in RAM rather than in flash. Fixed.